### PR TITLE
use rsplitn for rpc info extraction in case passwords have @

### PR DESCRIPTION
**Pull request by radixrat**
- Noticed I was unable to connect to RPC server due to `@` being a character in password, the current behaviour will split n times depending on how many of the @ special character appears and then will fail on URL since there are not two vectors.
- resolved this by splitting from right side and then swapping the vectors so no other code needed to be modified.

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -45,11 +45,12 @@ pub(crate) fn parse_startup_args() -> Result<LdkUserInfo, ()> {
 		return Err(());
 	}
 	let bitcoind_rpc_info = env::args().skip(1).next().unwrap();
-	let bitcoind_rpc_info_parts: Vec<&str> = bitcoind_rpc_info.split("@").collect();
+	let mut bitcoind_rpc_info_parts: Vec<&str> = bitcoind_rpc_info.rsplitn(2,"@").collect();
```
Comment (TheBlueMatt) [inline src/cli.rs:48]: Here and on the line below, `rustfmt` demands a space after the `,` in the parameter list.

File: src/cli.rs [hunk 2]
```diff
@@ -45,11 +45,12 @@ pub(crate) fn parse_startup_args() -> Result<LdkUserInfo, ()> {
...
 	if bitcoind_rpc_info_parts.len() != 2 {
 		println!("ERROR: bad bitcoind RPC URL provided");
 		return Err(());
 	}
+	bitcoind_rpc_info_parts.swap(0,1);
```
Comment (TheBlueMatt) [inline src/cli.rs:53]: nit: maybe more readable to just change the two lines that access `bitcoind_rpc_info_parts`?

File: src/cli.rs (hunk 1 above)
Comment (radixrat) [reply src/cli.rs]: didn't know about this tool yet, thanks, should be fixed in the latest commit

File: src/cli.rs (hunk 2 above)
Comment (radixrat) [reply src/cli.rs]: good suggestion, I tried this originally but wasn't sure if the reverse index access would be confusing, sounds like this is the better approach.

**Comment by radixrat** [comment]
> Can you squash the commits down into one commit to keep the git history clean?

should be squashed now.

**Review by TheBlueMatt — APPROVED** [review]

**Pull request merged**